src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -510,27 +510,28 @@ protected override bool TryExec(Guid commandGroup, uint commandId)
         {
             try
             {
-                // Let the presented know we're starting a search.  It will give us back
-                // the context object that the FAR service will push results into.
+                // Let the presented know we're starting a search.  It will give us back the context object that the FAR
+                // service will push results into.  This operation is not externally cancellable.  Instead, the find
+                // refs window will cancel it if another request is made to use it.
                 var context = presenter.StartSearch(
-                    EditorFeaturesResources.Find_References, supportsReferences: true);
+                    EditorFeaturesResources.Find_References, supportsReferences: true, CancellationToken.None);
 
-                var cancellationToken = context.CancellationToken;
-
-                // Kick off the work to do the actual finding on a BG thread.  That way we don'
-                // t block the calling (UI) thread too long if we happen to do our work on this
-                // thread.
-                await Task.Run(async () =>
+                try
+                {
+                    var cancellationToken = context.CancellationToken;
+
+                    // Kick off the work to do the actual finding on a BG thread.  That way we don'
+                    // t block the calling (UI) thread too long if we happen to do our work on this
+                    // thread.
+                    await Task.Run(async () =>
+                    {
+                        await FindReferencesAsync(symbolListItem, project, context).ConfigureAwait(false);
+                    }, cancellationToken).ConfigureAwait(false);
+                }
+                finally
                 {
-                    await FindReferencesAsync(symbolListItem, project, context).ConfigureAwait(false);
-                }, cancellationToken).ConfigureAwait(false);
-
-                // Note: we don't need to put this in a finally.  The only time we might not hit
-                // this is if cancellation or another error gets thrown.  In the former case,
-                // that means that a new search has started.  We don't care about telling the
-                // context it has completed.  In the latter case something wrong has happened
-                // and we don't want to run any more code in this particular context.
-                await context.OnCompletedAsync().ConfigureAwait(false);
+                    await context.OnCompletedAsync().ConfigureAwait(false);
```

Comment:
Can we just make FindUsagesContext disposable? Seems like it would be more obvious what needs to happen with it, rather than having to know to call OnCompletedAsync.

Reply:
1. It's an existing api. 
2. Being disposable seems weird to me. We're not disposing it. We're just telling it that the search is over so it can stop showing results. 
3. It actually is disposable. But it's disposed when the window is actually closed. 

I did consider this, and it would make for a nicer using block, but it honestly felt *weird...* To me :-)